test/framework/resources/k8s/resources/job.go

```diff
@@ -59,3 +61,21 @@ func (d *defaultJobManager) CreateAndWaitTillJobCompleted(job *v1.Job) (*v1.Job,
 		return false, nil
 	}, ctx.Done())
 }
+
+func (d *defaultJobManager) DeleteAndWaitTillJobIsDeleted(job *v1.Job) error {
+	ctx := context.Background()
+	err := d.k8sClient.Delete(ctx, job)
+	if err != nil {
+		return err
+	}
+	observedJob := &v1.Job{}
+	return wait.PollImmediateUntil(utils.PollIntervalShort, func() (bool, error) {
```

Comment:
Is there a reason we need to wait for the job to be deleted?

Reply:
This is to cleanly transition to new test case. For instance once the new test is running the older Job's pod will not be in Running/Terminating state so it will not influence the results of the new test cases.

Reply:
Sounds good :)